### PR TITLE
Document equality

### PR DESCRIPTION
**Pull request by andrejbauer**

### CODE REVIEW

File: meta-language.md [hunk 1]
```diff
@@ -696,16 +699,81 @@ In the inferring mode the judgment that is being computed is unrestrained.
 
 In checking mode there is a given type $\tyA$ (actually a judgment $\istype{\G}{\tyA}$) and
 the judgment that is being computed must have the form $\isterm{\D}{\e}{\tyA}$. In other
-words, the type is prescribed in advanced. For example, an application
+words, the type is prescribed in advanced. For example, an equality type
```
Comment (SkySkimmer) [inline meta-language.md]: "in advanced"

File: meta-language.md [hunk 2]
```diff
@@ -696,16 +699,81 @@ In the inferring mode the judgment that is being computed is unrestrained.
...
+
+* Evaluate `c₁` in inferring mode to a judgement `⊢ e₁ : A`.
+* if `A` is syntactically equal to a product `∏ (x : B), C`, evaluate `c₂` in checking mode at type `B` to a judgement `⊢ e₂ : B`. In this case the result is `⊢ e₁ e₂ : C[e₂/x]`.
+* otherwise, trigger the operation `coerce_fun (⊢ e₁ : A)`.
+
+The user provided handler must yield a value of type `coercible` as follows:
+
+* `NotCoercible` if `e₁` is to be considered as not coercible to a product type (which results in
+  a runtime error),
+* `Convertible (⊢ ξ : A ≡ ∏ (x : B), C)` if th types `A` is equal to the product `∏ (x :
```
Comment (SkySkimmer) [inline meta-language.md]: "th types" -> "the type"

File: meta-language.md [hunk 3]
```diff
@@ -865,34 +933,138 @@ The equality type is computed with
...
 
-#### Reflexivity
+###### Reflexivity
 
 The reflexivity term is computed with
 
     refl c
 
-#### Equality checking
+If `c` evaluates to $\isterm{\G}{\e}{\tyA}$ then `refl c` evaluates to $\isterm{\G}{\juRefl{\tyA} \e}{\JuEqual{\tyA}{\e}{\e}}$.
```
Comment (SkySkimmer) [inline meta-language.md]: refl has special behavior in checking mode

File: meta-language.md [hunk 4]
```diff
@@ -865,34 +933,138 @@ The equality type is computed with
...
+The rule [`prod-beta`](./type-theory.html#prod-beta) is available through
 
-TODO: Describe `congruence`
+    beta_step x A B e₁ e₂
 
-##### Extensionality
+where:
 
-TODO: Describe `extensionality`
+* `A` evaluates to a type $\istype{\G}{\tyA}$
```
Comment (SkySkimmer) [inline meta-language.md]: No.
x evaluates to x : A1
A evaluates to evidence that A1 = A2
B evaluates to evidence that x : A1 |- B1 = B2
e1 evaluates to x : A1 |- e1 : B1
e2 evaluates to |- e2 : B2
(no checking mode is used to evaluate e1 and e2 unlike what might be expected)

then we get (lambda x : A1, e1) @[x : A2, B2] e2 = e1[e2/x]

**Comment by andrejbauer** [comment]
Oops, should I fix these or are you?

**Comment by SkySkimmer** [comment]
You do it.

**Pull request merged**